pkg/syncer/pv_to_backingdiskobjectid_mapping.go

```diff
@@ -0,0 +1,179 @@
+		},
+	}
+
+	querySelection := cnstypes.CnsQuerySelection{
+		Names: []string{
+			string(cnstypes.QuerySelectionNameTypeBackingObjectDetails),
+		},
+	}
+	queryAllResult, err := metadataSyncer.volumeManager.QueryAllVolume(ctx, queryFilter, querySelection)
+	if err != nil {
+		log.Errorf("csiGetPVtoBackingDiskObjectIdMapping: failed to QueryAllVolume with err=%+v", err.Error())
+		return
+	}
+
+	// Get K8s PVs in State "Bound".
+	k8sPVs, err := getBoundPVs(ctx, metadataSyncer)
+	if err != nil {
+		log.Errorf("csiGetPVtoBackingDiskObjectIdMapping: Failed to get PVs from kubernetes. Err: %+v", err)
+		return
+	}
+
+	// volumeHandleToPvcMap maps pv.Spec.CSI.VolumeHandle to the pvc object which
+	// bounded to the pv.
+	volumeHandleToPvcMap := make(volumeHandlePVCMap, len(k8sPVs))
+	// volumeHandleToPvUidMap maps volumehandle to pv uid
+	volumeHandleToPvUidMap := make(volumeIdToPvUidMap, len(k8sPVs))
+
+	for _, pv := range k8sPVs {
+		if pv.Spec.ClaimRef != nil && pv.Status.Phase == v1.VolumeBound {
+			pvc, err := metadataSyncer.pvcLister.PersistentVolumeClaims(
+				pv.Spec.ClaimRef.Namespace).Get(pv.Spec.ClaimRef.Name)
+			if err != nil {
+				log.Warnf("csiGetPVtoBackingDiskObjectIdMapping: Failed to get pvc for namespace %s and name %s. err=%+v",
+					pv.Spec.ClaimRef.Namespace, pv.Spec.ClaimRef.Name, err)
+				continue
+			}
+			volumeHandleToPvcMap[pv.Spec.CSI.VolumeHandle] = pvc
+			log.Debugf("csiGetPVtoBackingDiskObjectIdMapping: pvc %s/%s is backed by pv %s volumeHandle %s",
+				pvc.Namespace, pvc.Name, pv.Name, pv.Spec.CSI.VolumeHandle)
+			volumeHandleToPvUidMap[pv.Spec.CSI.VolumeHandle] = string(pv.UID)
+			log.Debugf("csiGetPVtoBackingDiskObjectIdMapping: volumehandle %s mapping to pv uid %s",
+				pv.Spec.CSI.VolumeHandle, string(pv.UID))
+		}
+	}
+
+	// pv to backingDiskObjectId Map maps vol.VolumeId.Id to backingobjectId.
+	volumeIdToBackingObjectIdMap := make(volumeIdToPvUidMap, len(queryAllResult.Volumes))
+
+	for _, vol := range queryAllResult.Volumes {
+		volumeIdToBackingObjectIdMap[vol.VolumeId.Id] = vol.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).BackingDiskId
+	}
+
+	for volID, pvc := range volumeHandleToPvcMap {
+		pvToBackingDiskObjectIdPair := volumeHandleToPvUidMap[volID] + ":" + volumeIdToBackingObjectIdMap[volID]
+		updated, err := updatePVtoBackingDiskObjectIdMappingStatus(ctx, k8sclient, pvc, pvToBackingDiskObjectIdPair)
+		if err != nil {
+			log.Error("csiGetPVtoBackingDiskObjectIdMapping: Failed to update pv to backingDiskObjectId mapping")
+			return
+		}
+		if updated {
+			log.Debugf("csiGetPVtoBackingDiskObjectIdMapping: pvc %s is updated with pv to backingDiskObjectId mapping %s", pvc.Name, pvToBackingDiskObjectIdPair)
+		}
+	}
+
+	log.Debug("csiGetPVtoBackingDiskObjectIdMapping: end")
+}
+
+func validatePvToBackingDiskObjectIdPair(pvToBackingDiskObjectIdPair string) bool {
+	split := strings.Split(pvToBackingDiskObjectIdPair, ":")
+	return len(split) == 2
```

Comment:
This will always return true, you are constructing the string in line 94 with a delimiter ":", even if `volumeIdToBackingObjectIdMap[volID]` is an empty string, it'll still be able to split into two different strings, with one of them being an empty string ""

Reply:
Updated validatePvToBackingDiskObjectIdPair to check split string content rather than length.